diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0cbd8e5989..609c5b166f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -634,7 +634,8 @@ if(CONFIG_BUILD_OUTPUT_STRIP_PATHS)
   zephyr_cc_option(-fmacro-prefix-map=${CMAKE_SOURCE_DIR}=CMAKE_SOURCE_DIR)
   zephyr_cc_option(-fmacro-prefix-map=${ZEPHYR_BASE}=ZEPHYR_BASE)
   if(WEST_TOPDIR)
-    zephyr_cc_option(-fmacro-prefix-map=${WEST_TOPDIR}=WEST_TOPDIR)
+    file(TO_CMAKE_PATH "${WEST_TOPDIR}" WEST_TOPDIR_FWD)
+    zephyr_cc_option(-fmacro-prefix-map=${WEST_TOPDIR_FWD}=WEST_TOPDIR)
   endif()
 endif()
 
